src/mesh_motion/MeshMotionAlg.C

```diff
@@ -84,6 +85,23 @@ void MeshMotionAlg::initialize( const double time )
     frameVec_[i]->update_coordinates_velocity(time);
   }
 
+  // TODO: NGP Transition
+  // Manually synchronize fields to device
+  {
+    const auto& meta = bulk_.mesh_meta_data();
+    std::vector<std::string> fnames{
+      "current_coordinates",
+      "mesh_displacement",
+      "mesh_velocity",
+    };
+
+    for (const auto& ff: fnames) {
+      auto* fld = meta.get_field(stk::topology::NODE_RANK, ff);
```

Comment:
I always have an internal alarm bell go off when I see get-field-by-name. Perhaps this isn't done often enough to be a performance issue. But don't we have the actual fields somewhere nearby?

Reply:
This class doesn't hold the fields, this is more like the AlgDrivers we have been using. It is holds the sub-algorithms that hold references to the field.